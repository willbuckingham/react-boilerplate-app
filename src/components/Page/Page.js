import React from 'react'
import propTypes from 'prop-types'
import {
  AppBar,
  Toolbar,
  IconButton,
  // Typography,
  // Button,
} from '@material-ui/core'
import { makeStyles } from '@material-ui/core/styles'
import {
  Menu,
} from '@material-ui/icons'

const useStyles = makeStyles(({ spacing }) => ({
  root: {
    flexGrow: 1,
  },
  menuButton: {
    marginRight: spacing(2),
  },
  title: {
    flexGrow: 1,
  },
  main: {
    margin: spacing(2),
  },
}))

const Page = ({ children }) => {
  const classes = useStyles()

  return (
    <div>
      <AppBar position='static'>
        <Toolbar>
          <IconButton edge='start' className={classes.menuButton} color='inherit' aria-label='menu'>
            <Menu />
          </IconButton>
        </Toolbar>
      </AppBar>

      <main className={classes.main}>
        { children }
      </main>
    </div>
  )
}

Page.propTypes = {
  children: propTypes.node.isRequired,
}

export default Page
