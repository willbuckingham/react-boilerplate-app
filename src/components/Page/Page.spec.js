import React from 'react'
import { shallow } from 'enzyme'

import Page from './Page'

describe('Page', () => {
  it('renders', () => {
    expect(shallow(<Page>child</Page>)).toMatchSnapshot()
  })
})
