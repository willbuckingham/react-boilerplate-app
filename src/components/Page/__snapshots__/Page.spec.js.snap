// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page renders 1`] = `
<div>
  <WithStyles(ForwardRef(AppBar))
    position="static"
  >
    <WithStyles(ForwardRef(Toolbar))>
      <WithStyles(ForwardRef(IconButton))
        aria-label="menu"
        className="makeStyles-menuButton-2"
        color="inherit"
        edge="start"
      >
        <MenuIcon />
      </WithStyles(ForwardRef(IconButton))>
    </WithStyles(ForwardRef(Toolbar))>
  </WithStyles(ForwardRef(AppBar))>
  <main
    className="makeStyles-main-4"
  >
    child
  </main>
</div>
`;
